### PR TITLE
deps: update to uefi-rs 0.21.0

### PR DESCRIPTION
**Pull request by RaitoBezarius**
It simplifies our filesystem handling.

### CODE REVIEW

File: rust/stub/Cargo.toml [hunk 1]
```diff
@@ -7,13 +7,12 @@ publish = false
...
-uefi = { version = "0.20.0", default-features = false, features = [ "alloc", "global_allocator" ] }
-uefi-services = { version = "0.17.0", default-features = false, features = [ "panic_handler", "logger" ] }
+uefi = { version = "0.21.0", default-features = false, features = [ "alloc", "global_allocator" ] }
+uefi-services = { version = "0.18.0", default-features = false, features = [ "panic_handler", "logger" ] }
 goblin = { version = "0.6.1", default-features = false, features = [ "pe64", "alloc" ]}
 bitflags = "2.2.1"
 
-# Even in debug builds, we don't enable the debug logs, because they generate a lot of spam from goblin.
-log = { version = "0.4.17", default-features = false, features = [ "max_level_info", "release_max_level_warn" ]}
+log = { version = "0.4.17", default-features = false, features = [ "max_level_info", "release_max_level_info" ]}
```
Comment (RaitoBezarius) [inline rust/stub/Cargo.toml:15]: ignore this change

File: rust/stub/src/main.rs [hunk 2]
```diff
@@ -198,33 +195,10 @@ fn main(handle: Handle, mut system_table: SystemTable<Boot>) -> Status {
...
-            )
-            .expect("Failed to open initrd for reading")
-            .into_regular_file()
-            .expect("Initrd is not a regular file");
-
-        initrd_data = read_all(&mut initrd_file).expect("Failed to read kernel file into memory");
+
+
+        kernel_data = file_system.read(&*config.kernel_filename).expect("Failed to read kernel file into memory");
+        initrd_data = file_system.read(&*config.initrd_filename).expect("Failed to read kernel file into memory");
```
Comment (nikstur) [inline rust/stub/src/main.rs:201]:
```suggestion
        initrd_data = file_system.read(&*config.initrd_filename).expect("Failed to read initrd file into memory");
```

**Review by nikstur — COMMENTED** [review]
Looks good. What needs to be done to finish this?

**Comment by RaitoBezarius** [comment]
undoing my fucking changes

File: rust/stub/src/main.rs (hunk 2 above)
Comment (RaitoBezarius) [reply rust/stub/src/main.rs:201]: will do

**Comment by RaitoBezarius** [comment]
@nikstur ready

File: rust/stub/Cargo.toml [hunk 3]
```diff
@@ -7,12 +7,11 @@ publish = false
...
 
 [dependencies]
-uefi = { version = "0.20.0", default-features = false, features = [ "alloc", "global_allocator" ] }
-uefi-services = { version = "0.17.0", default-features = false, features = [ "panic_handler", "logger" ] }
+uefi = { version = "0.21.0", default-features = false, features = [ "alloc", "global_allocator" ] }
+uefi-services = { version = "0.18.0", default-features = false, features = [ "panic_handler", "logger" ] }
 goblin = { version = "0.6.1", default-features = false, features = [ "pe64", "alloc" ]}
 bitflags = "2.2.1"
 
-# Even in debug builds, we don't enable the debug logs, because they generate a lot of spam from goblin.
```
Comment (nikstur) [inline rust/stub/Cargo.toml:15]: Was this deleted by mistake? Doesn't really matter much.

**Review by nikstur — APPROVED** [review]
LGTM

File: rust/stub/Cargo.toml (hunk 3 above)
Comment (RaitoBezarius) [reply rust/stub/Cargo.toml:15]: Yes, will restore it

**Pull request merged**